### DISCUSSION TITLE
Modification in Default Manage Content Grid View Design

### DISCUSSION THREAD

**Discussion by Syed-Hasnain-Askari** [OrchardCMS/OrchardCore#12350] [Q&A] [answered]
Hey,

The default grid view design of  Manage Content in Orchard that is,

![image](https://user-images.githubusercontent.com/47834896/188972496-58891732-3c9b-4e9e-9a12-d542f352a3d7.png)

but I want to modify it like this one,

![image_2022_09_07T20_29_27_870Z](https://user-images.githubusercontent.com/47834896/188973621-5a34b21c-c7d1-4525-81f5-af41df98167c.png)

Is there's any possible solution for it?  I will be grateful for any help you can provide. @Skrypt @ns8482e @lampersky

**Comment by Skrypt** [comment]
I think you will need to override the current OrchardCore.Contents modules razor view for achieving that. Though, I can't guarantee that we won't break anything in the future if you go that path. The other option is to copy the entire OrchardCore.Contents module and to rename it and customize it for your own needs.

**Reply by saleemhaq** [reply]
@Skrypt we have n number of content type and every content type have different fields so we need to show that like in content type register we need to show(firstname,lastname,email) for conact us need to show (email, postal address, mobile number) so how we achieve in that scenario please advise @Skrypt @ns8482e @hishamco @sebastienros

**Reply by Skrypt** [reply]
Same thing, you can create specific admin razor templates for each of these content types.

**Reply by Skrypt** [reply]
Content-ContentTypeName.SummaryAdmin.cshtml

https://docs.orchardcore.net/en/dev/docs/glossary/#display-type

**Comment by ns8482e** [comment, marked as answer]
@Syed-Hasnain-Askari  currently there is no shared/single view that you can override. However for Cotent Items list you can override ContentsAdminList view

**Reply by saleemhaq** [reply]
@ns8482e can you please share the sample code or example so I will achieve thanks for you support in advance. @Skrypt

**Reply by ns8482e** [reply]
I don’t have single ready to use answer for you

As @Skrypt suggest, you can leverage and override  SummaryAdmin for your content item

List(ul) and list items(li) are rendered by ContentsAdminList view. You can dig up more In source of OrchardCore.Contents module

**Reply by Syed-Hasnain-Askari** [reply]
Hey folks thanks for your response,

I have modified it like this

![image](https://user-images.githubusercontent.com/47834896/189433018-0082eae9-c7b5-4e84-a8c5-db12b5680195.png)

but I how can I design it for particular content Items?

I'm unable to understand Content-ContentTypeName.SummaryAdmin.cshtml structure, could please give some code snippets for it?
@ns8482e @Skrypt

**Reply by hishamco** [reply]
You could check our docs

**Reply by ns8482e** [reply]
@Syed-Hasnain-Askari If I were you I’d have override ContentsAdminList such a way that  each list item (li) define different  zones(that represent each column needed eg Content, Column1, Column2) then for each content item summary, using placement rules I’d place field to a zone that defined for the given column
https://docs.orchardcore.net/en/latest/docs/reference/core/Placement

**Reply by Syed-Hasnain-Askari** [reply]
```C#
@if (Model.ContentItems.Count > 0)
    {
    @foreach (var contentItemSummary in Model.ContentItems)
    {
        if (contentItemSummary.ContentItem.ContentType == "contactus")
        {

        <li>@contentItemSummary.ContentItem.Content.contactus.FirstName.Text</li>
        <li>@contentItemSummary.ContentItem.Content.contactus.Pharmacyname.Text</li>
        <li>@contentItemSummary.ContentItem.Content.contactus.Email.Text</li>
        }
else
{
        @await DisplayAsync(contentItemSummary)
}
}
    
 }
```

This work for me but the issue is that I need all Buttons and checkbox which comes from   `@await DisplayAsync(contentItemSummary)` method

![image](https://user-images.githubusercontent.com/47834896/189483002-3f26f168-d080-4533-be3e-32c125697871.png)

How can I override `DisplayAsync(contentItemSummary)` for my purpose?  I will be grateful for any help you can provide. @ns8482e

**Reply by ns8482e** [reply]
You don’t need to change most of ContentsAdminList, you override just to add custom things like table column header.

you always render using                 `@await DisplayAsync(contentItemSummary)`

Content item that needs column like display you override that content item’s summaryadmin shape and place content fields that needs under specific column using placement

**Reply by ns8482e** [reply]
See documentation https://docs.orchardcore.net/en/latest/docs/reference/modules/Templates/#templates-orchardcoretemplates for creating different templates for content types

**Reply by Skrypt** [reply]
Please @Syed-Hasnain-Askari use proper Github markdown when sharing blocks of code by using this method: 

https://docs.github.com/en/get-started/writing-on-github/getting-started-with-writing-and-formatting-on-github/basic-writing-and-formatting-syntax#quoting-code

**Reply by Syed-Hasnain-Askari** [reply]
I got my solution. Thanks a lot!  @ns8482e @Skrypt
